### DISCUSSION TITLE
CSV Import of all historic data - (training history/calendar, weight, sets/reps, calories etc.)

### DISCUSSION THREAD

**Discussion by nodecentral** [wger-project/wger#1202] [Q&A] [answered]
Hi

I’m looking to make the jump to Wger and would like to bring over all my historic fitness and health data. I can see in the read me that it references an import feature via csv but I can’t see any details ?    All help appreciated..

**Comment by rolandgeider** [comment]
Hi! There is a CSV import for (body) weight entries. You could get the rest into the system by talking to the system via the API and perhaps manually adding some workouts and such, but depending on what exactly you have this could be quite a bit of work

**Comment by nodecentral** [comment]
Thanks, that helps, and I’ve just had a quick look at the REST API (https://wger.de/en/software/api, although I can’t seem to see something with specific json examples for the required POST format etc.?)

As I have some basic programming skills with Lua, I might be able to work something out that could loop through all my data and post it via the api.

**Reply by rolandgeider** [reply]
yeah, I've been meaning to add some proper documentation to the API for a while, but you know... documentation 😅 

the easiest way is to add a couple of weight entries, a  simple workout, some logs, etc. and then take a look at the API so you know what to POST

**Reply by rolandgeider** [reply]
if it helps I can point you to the places where we use it in the flutter app or the react bits that we have

**Reply by nodecentral** [reply]
Hi @rolandgeider - if you are able to share some real world examples of a few jsons used to do a POST (suggestions below), that would be great, I can build from there. 

(i) json for a multi set/rep workout on a specific date/time
(ii) json for an update to my weight on a specific date/time
(iii) json for a meal eaten on a specific date/time
Etc.

**Comment by nodecentral** [comment, marked as answer]
Will pick this up under this discussion thread..

https://github.com/wger-project/wger/discussions/1232
